### PR TITLE
Updates to README and notebook for Reranker 

### PR DESCRIPTION
**Pull request by bhavani105**
# Pull Request      

## What does this PR do?

1. Move common arguments to Reranker base class
2. Add Reranker documentation  to components/README.md  and services/README.md 
3. Add reranking step using ColBERTReranker to notebooks/retriever-reader-pipelines/extractive_reader_with_discovery.ipynb

Notes:
- Replace `(issue)` **above** ↑↑↑ with the issue this PR closes to automatically link the two.
This must be done when the PR is created.
- Add multiple `Closes #(issue)` as needed.
- If this PR is work towards but does not close an issue, simply tag the issue without mentioning `Closes`.

## Description

**Describe** the changes proposed by this PR below to give the reviewer context below ↓↓↓

```(description)```     

### Request Review

Be sure to **request a review** from one or more reviewers (unless the PR is to an unprotected branch).

## Versioning

When opening a PR to make changes to PrimeQA (i.e. `primeqa/`) master, be sure to increment the version following
[semantic versioning](https://semver.org/).  The VERSION is stored [here](https://github.com/primeqa/primeqa/blob/main/VERSION) 
and is incremented using `bump2version {patch,minor,major}` as described in the (development guide documentation)[development.html].

 - [ ] Have you updated the VERSION?
 - [ ] Or does this PR not change the `primeqa` package or was not into master?

After pulling in changes from master to an existing PR, ensure the VERSION is updated appropriately.
This may require bumping the version again if it has been previously bumped.

If you're not quite ready yet to post a PR for review, feel free to open a [draft PR](https://github.blog/2019-02-14-introducing-draft-pull-requests/).

### After Merging
If merging into master and VERSION was updated, after this PR is merged:
- [ ] [Create a release](https://docs.github.com/en/github/administering-a-repository/releasing-projects-on-github/managing-releases-in-a-repository#creating-a-release) from the master with version equal to [VERSION](https://github.com/primeqa/primeqa/blob/main/VERSION)
- [ ] Not merging to master or VERSION not updated

### Checklist

Review the following and mark as completed:
- [ ] [Tag an issue or issues](#what-does-this-pr-do) this PR addresses.
- [ ] Added [description](#description) of changes proposed.
- [ ] Review requested as [appropriate](#request-review).
- [ ] Version bumped as [appropriate](#Versioning).
- [ ] New classes, methods, and functions documented.
- [ ] Documentation for modified code is updated.
- [ ] Built documentation to confirm it renders as expected (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Tests added to ensure all functionalities tested at >= 60% unit test coverage (see [here](development.md)).
- [ ] Code cleaned up and commented out code removed.
- [ ] Release created [as needed](#after-merging) after merging.

### CODE REVIEW

File: primeqa/components/reranker/colbert_reranker.py [hunk 1]
```diff
@@ -14,8 +14,8 @@ class ColBERTReranker(BaseReranker):
 
     Args:
         model (str, optional): Model to load.
-        collection (str, optional): collection to load. Defaults to collection in index configuration.
-        max_num_documents (int, optional): Maximum number of reranked document to return. Defaults to 5.
+        max_num_documents (int, optional): Maximum number of reranked document to return. Defaults to -1.
+        include_title (bool, optional): Whether to concate text and title. Defaults to True
```
Comment (jdpsen) [inline primeqa/components/reranker/colbert_reranker.py:18]: whether to "concatenate" ?

File: primeqa/components/reranker/seq_classification_reranker.py [hunk 2]
```diff
@@ -17,8 +17,11 @@ class SeqClassificationReranker(BaseReranker):
 
     Args:
         model (str, optional): Model to load. 
-        collection (str, optional): collection to load. Defaults to collection in index configuration.
-        max_num_documents (int, optional): Maximum number of reranked document to return. Defaults to 5.
+        max_num_documents (int, optional): Maximum number of reranked document to return. Defaults to -1.
+        include_title (bool, optional): Whether to concate text and title. Defaults to True
```
Comment (jdpsen) [inline primeqa/components/reranker/seq_classification_reranker.py:21]: Whether to "concatenate" ?

**Review by jdpsen — APPROVED** [review]
minor suggestions for grammar corrections. 
LGTM otherwise.

File: primeqa/services/README.md [hunk 3]
```diff
@@ -100,23 +103,26 @@ docker build -f Dockerfiles/Dockerfile.cpu -t primeqa:$(cat VERSION) --build-arg
 ```
 <h4> Run Container </h4>
 
-The container needs write access to a cache directory for caching Huggingface model and datasets.  Additionally will need write access to a store directory for custom models and index creation. See [Store](./store)
+The container needs write access to a `cache` directory e.g. `$HOME/.cache/` for caching Huggingface model and datasets.  Additionally will need write access to a `store` directory, e.g. `$PWD/store/` for custom models and index creation.
```
Comment (franzmpub) [inline primeqa/services/README.md:106]: Additionally it will need write access

**Review by franzmpub — APPROVED** [review]
LGTM, suggested one trivial README change

**Pull request merged**